### PR TITLE
Finals application refactoring

### PR DESCRIPTION
**Pull request by RiccardoGrieco**
This PR to refactor the applications for the XPrize finals.

cc @lrapetti @CarlottaSartore @mebbaid  @DanielePucci  @GianlucaMilani @paolo-viceconte @isorrentino

### CODE REVIEW

**Comment by RiccardoGrieco** [comment]
This should follow https://github.com/ami-iit/component_ANA-Avatar-XPRIZE/issues/766

**Comment by RiccardoGrieco** [comment]
We tested it with @lrapetti @mebbaid  @CarlottaSartore @GianlucaMilani @DanielePucci and it seemed to be working fine.

⚠️ This was tested without the operator side

**Comment by GiulioRomualdi** [comment]
Hi @RiccardoGrieco I would also add the logging, it is useful to have it if we want to run it to collect some data. You can move it at the end of the step 4

File: app/scripts/Xprize-Finals-4.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    <name>HapticGloveModule</name>
+    <node>icub-head</node>
+  </module>
+
+  <module>
+    <name>WalkingModule</name>
+    <node>icub-head</node>
+    <parameters>--from dcm_walking_iFeel_joint_retargeting.ini</parameters>
+  </module>
+
```
Comment (GiulioRomualdi) [inline app/scripts/Xprize-Finals-4.xml:24]:
```suggestion

  <module>
    <name>yarprobotinterface</name>
    <parameters>--config launch-yarp-robot-logger.xml</parameters>
    <node>icub-console</node>
    <workdir>/home/icub/Desktop/walking_logger</workdir>
  </module>

```

**Review by S-Dafarra — APPROVED** [review]

**Pull request merged**